apps/api/src/app/vercel-integration/usecases/setup-integration/setup-integration.usecase.ts

```diff
@@ -0,0 +1,149 @@
+
+      return {
+        success: true,
+      };
+    } catch (error) {
+      throw new ApiException(
+        error?.response?.data?.error_description || error?.response?.data?.message || error.message
+      );
+    }
+  }
+
+  async getVercelToken(code: string) {
+    const postData = new URLSearchParams({
+      code,
+      client_id: process.env.VERCEL_CLIENT_ID || 'oac_wwSTKHT6nQ2hdosgHavoOG2U',
+      client_secret: process.env.VERCEL_CLIENT_SECRET || 'NLuf2j7avAVJLh8VhLzxs4M1',
+      redirect_uri: process.env.VERCEL_REDIRECT_URI || 'http://localhost:4200/auth/login',
+    });
+    const response = await lastValueFrom(
+      this.httpService.post(`${process.env.VERCEL_BASE_URL}/v2/oauth/access_token`, postData, {
+        headers: {
+          'Content-Type': 'application/x-www-form-urlencoded',
+        },
+      })
+    );
+    const data = await response.data;
+
+    return {
+      accessToken: data.access_token,
+      userId: data.user_id,
+      teamId: data.team_id,
+    };
+  }
+
+  async getVercelProjects(token: string, teamId: string) {
+    const response = await lastValueFrom(
+      this.httpService.get(`${process.env.VERCEL_BASE_URL}/v4/projects${teamId ? `?teamId=${teamId}` : ''}`, {
+        headers: {
+          Authorization: `Bearer ${token}`,
+        },
+      })
+    );
+
+    return response.data.projects;
+  }
+
+  async getEnvKeys(command: SetupIntegrationCommand) {
+    const apiKeysResponse = await this.getApiKeyUsecase.execute(
+      GetApiKeysCommand.create({
+        environmentId: command.environmentId,
+        organizationId: command.organizationId,
+        userId: command.userId,
+      })
+    );
+
+    const envResponse = await this.getEnvironmentUsecase.execute(
+      GetEnvironmentCommand.create({
+        organizationId: command.organizationId,
+        userId: command.userId,
+        environmentId: command.environmentId,
+      })
+    );
+
+    return {
+      privateKey: apiKeysResponse[0].key,
+      clientKey: envResponse.identifier,
+    };
+  }
+
+  async setEnvironments({ identifier, projectIds, secretKey, teamId, token }: ISetEnvironment) {
+    const projectApiUrl = `${process.env.VERCEL_BASE_URL}/v9/projects`;
+
+    const target = ['production', 'preview', 'development'];
+    const type = 'encrypted';
+
+    const apiKeys = [
+      {
+        target,
+        type,
+        value: identifier,
+        key: 'NOVU_CLIENT_APP_ID',
+      },
+      {
+        target,
+        type,
+        value: secretKey,
+        key: 'NOVU_API_SECRET',
+      },
+    ];
+
+    await Promise.all(
+      projectIds.map((projectId) =>
+        lastValueFrom(
+          this.httpService.post(`${projectApiUrl}/${projectId}/env${teamId ? `?teamId=${teamId}` : ''}`, apiKeys, {
+            headers: {
+              Authorization: `Bearer ${token}`,
+              'Content-Type': 'application/json',
+            },
+          })
+        )
+      )
+    );
+  }
+}
+
+interface ISetEnvironment {
```

Comment:
This should be placed between the imports and beginning the class instead of the bottom of this file